src/bin/pg_autoctl/cli_drop_node.c

```diff
@@ -578,9 +578,44 @@ cli_drop_local_node(KeeperConfig *config, bool dropAndDestroy)
 	 */
 	pid_t pid = 0;
 
-	if (!wait_for_pid_to_exit(config->pathnames.pid, 30, &pid))
+	/*
+	 * Before continuing we need to make sure that a currently running service
+	 * has stopped.
+	 *
+	 * If --force isn't used then a running pg_autoctl process will detect that
+	 * it is dropped and clean itself up nicely and finally it will exit. We
+	 * give the process 30 seconds to exit by itself.
+	 *
+	 * If --force is used, we skip the transition to "dropped". So a currently
+	 * running process won't realise it's dropped, so it will not exit by
+	 * itself. So there's no point in waiting for 30 seconds.
+	 */
+	bool stopped;
+	if (!dropForce)
+	{
+		if (!wait_for_process_to_stop(config->pathnames.pid, 30, &stopped, &pid))
+		{
+			/* errors have already been logged */
+			exit(EXIT_CODE_INTERNAL_ERROR);
+		}
+	}
+	else
+	{
+		if (!is_process_stopped(config->pathnames.pid, &stopped, &pid))
```

Comment:
The function `is_process_stopped` is used a lot and often with a negation before it. I think it would be all easier if we had the function `pidfile_process_is_running` (similar to `pg_setup_is_running` and others).

Reply:
Is discussed over chat, flipping this brings in other consistency problems. Leaving it like this.